### PR TITLE
Batch Beta Performance Improvements

### PR DESCRIPTION
**Pull request by annatisch**
Added container resource file cache

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@annatisch__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by jasper-schneider** [comment]
Looks fine to me, just need to fix some jshint errors.

**Comment by jasper-schneider** [comment]
Setting correct CLA label - Anna is an Azure FTE

File: lib/commands/batch/batch.fileUtils._js [hunk 1]
```diff
@@ -51,21 +52,49 @@ batchFileUtils.init = function (azureCli) {
...
     var operation = storageUtil.getStorageOperation(blobService, storageUtil.OperationType.Blob, 'listAllBlobs');
     var storageOptions = storageUtil.getStorageOperationDefaultOption();
-    storageOptions.prefix = prefix;
-    var contents = storageUtil.performStorageOperation(operation, _, container, storageOptions);
-    return contents;
+    var blobs = storageUtil.performStorageOperation(operation, _, container, storageOptions);
+    blobs.forEach_(_, 1, function(_, blob) {
+      var blobSas =  source.fileGroup ? batchFileUtils.generateSasToken(blob, container, blobService) : batchFileUtils.constructSasUrl(blob, url.parse(source.containerUrl));
+      var fileName = pathUtil.basename(blob.name);
+      var fileNameOnly = fileName.substr(0, fileName.lastIndexOf('.'));
```
Comment (xingwu1) [inline lib/commands/batch/batch.fileUtils._js]: This is a bug. The fileName may not contain '.', then it will throw exception.

Comment (annatisch) [inline lib/commands/batch/batch.fileUtils._js]: Oops! Sorry forgot about that

**Pull request merged**